apps/web/modules/event-types/views/event-types-listing-view.tsx

```diff
@@ -475,8 +476,11 @@ export const EventTypeList = ({
         {types.map((type, index) => {
           const embedLink = `${group.profile.slug}/${type.slug}`;
           const calLink = `${bookerUrl}/${embedLink}`;
-          const isPrivateURLEnabled = type.hashedLink?.link;
-          const placeholderHashedLink = `${WEBSITE_URL}/d/${type.hashedLink?.link}/${type.slug}`;
+          const isPrivateURLEnabled =
+            type.hashedLink && type.hashedLink.length > 0
+              ? type.hashedLink[privateLinkCopyIndices[type.slug] ?? 0]?.link
+              : false;
+          const placeholderHashedLink = `${WEBSITE_URL}/d/${isPrivateURLEnabled}/${type.slug}`;
```

Comment:
This doesn't make sense to me. Currently the type of the isPrivateURLEnabled is string | false, but it's used inside url also. Instead can you only store the link inside `const isPrivateURLEnabled = type.hashedLink[privateLinkCopyIndices[type.slug] ?? 0]?.link;`

Reply:
@anikdhabal Without the inline checks, i.e `type.hashedLink && type.hashedLink.length > 0`, it might throw errors when accessing array indices out-of-bounds.

But yes your point also makes sense that `bool` doesn't make any sense. Therefore I have converted the fallback value to an empty string. Does that make sense now. If the value is empty the`placeholderHashedLink` will be constructed with an empty `isPrivateURLEnabled`, but will not be displayed as per the below condition, therefore I think that should be fine. Please let me know your opinion.